src/features/metadata/stories/EnvBuilds.stories.tsx

```diff
@@ -1,27 +1,44 @@
-import type { Story, ComponentMeta } from "@storybook/react";
+import type { Meta, StoryObj } from "@storybook/react";
+import { useArgs } from "@storybook/preview-api";
+
 import { Provider } from "react-redux";
 import React from "react";
 
 import { store } from "../../../store";
-import { EnvBuilds } from "../components";
-import { mockBuilds } from "../mocks";
+import { EnvBuilds, IEnvBuildsProps } from "../components/EnvBuilds";
+import { mockBuilds } from "../mocks/mockBuilds";
 
 export default {
-  component: EnvBuilds
-} as ComponentMeta<typeof EnvBuilds>;
+  component: EnvBuilds,
+  decorators: [
+    Story => (
+      <Provider store={store}>
+        <Story />
+      </Provider>
+    )
+  ]
+} as Meta<typeof EnvBuilds>;
 
-export const Primary = {
+type Story = StoryObj<typeof EnvBuilds>;
+
+export const Primary: Story = {
   args: {
     currentBuildId: 1,
     selectedBuildId: 1,
     mode: "read-only",
     builds: mockBuilds
   },
-  decorators: [
-    (Story: Story) => (
-      <Provider store={store}>
-        <Story />
-      </Provider>
-    )
-  ]
+  render: function Render(args: IEnvBuildsProps) {
+    const [{ selectedBuildId = args.selectedBuildId }, updateArgs] = useArgs();
+
+    const handleChangeToSelectedBuildId = () => {
+      const { id: nextSelectedBuildId = args.selectedBuildId } =
+        store.getState().enviroments.currentBuild;
+      updateArgs({ selectedBuildId: nextSelectedBuildId });
+    };
+
+    store.subscribe(handleChangeToSelectedBuildId);
```

Comment:
These lines of code are the driving motivation upgrading Storybook.

The EnvBuilds component contains a BuildDropdown component. When the component renders as a story within Storybook with the default args defined above (lines 25-30), it presents a dropdown to the end user that does not change if the user actually clicks on the dropdown and selects one of the options. That's because Storybook provides its own UI (below the component) to update the story args. 

To solve this problem Storybook provides a React hook called [`useArgs()`](https://storybook.js.org/docs/writing-stories/args#setting-args-from-within-a-story). 

I tried following the docs not realizing that our version of Storybook was already 2 major versions behind. So of course it didn't work, and I kept running into hard-to-debug issues, so I figured it would be best for the health of the repo to upgrade Storybook.

Reply:
I'm a big fan of keeping up to date. Maintenance becomes incredibly hard to do if you fall way behind on major releases. Thanks for doing this!

Is there some way we can be tracking new major releases? Dependabot maybe? If you think it could be helpful, please make an issue for it.

Reply:
- https://github.com/conda-incubator/conda-store-ui/issues/388